nexsock-web/src-ts/core/web-components.ts

```
@@ -0,0 +1,124 @@
+/**
+ * Web Components bridge for JSX components
+ * Allows JSX components to be used directly in HTML templates as custom elements
+ */
+
+interface ComponentDefinition {
+  component: Function;
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)